client/src/main/scala/com/dwolla/cloudflare/domain/model/ratelimits/package.scala

```diff
@@ -0,0 +1,173 @@
+package com.dwolla.cloudflare.domain.model
+
+import com.dwolla.circe._
+import io.circe.{Decoder, Encoder}
+import shapeless.tag.@@
+
+package object ratelimits {
+
+  type RateLimitId = String @@ RateLimitIdTag
+
+  private[cloudflare] val tagRateLimitId: String ⇒ RateLimitId = shapeless.tag[RateLimitIdTag][String]
+
+}
+
+package ratelimits {
+  import java.time.Duration
+
+  import io.circe.ObjectEncoder
+  import io.circe.export.Exported
+  import io.circe.generic.auto._
+
+  trait RateLimitIdTag
+
+  case class RateLimit(id: Option[RateLimitId] = None,
+                       disabled: Option[Boolean] = None,
+                       description: Option[String] = None,
+                       `match`: RateLimitMatch,
+                       bypass: Option[Seq[RateLimitBypass]] = None,
+                       threshold: Int,
+                       period: Duration,
+                       action: RateLimitAction)
+
+  case class RateLimitMatch(request: RateLimitMatchRequest, response: Option[RateLimitMatchResponse] = None)
+
+  case class RateLimitMatchRequest(methods: Option[Seq[Method]] = None,
```

Comment:
For these types like `Option[Seq[?]]`, what is the difference in semantic meaning between `Some(Seq.empty)` and `None`?

Reply:
For case classes modeling Cloudflare JSON objects, any field with some type `Option[?]` indicates it is not guaranteed to show up as a field in the JSON response returned by Cloudflare. So, in the case of a field with type `Option[Seq[?]]`, `None` represents the case where the field was not returned by Cloudflare and `Some(Seq.empty)` the case where the field was returned but is an empty list.

Reply:
Do we care about that distinction? If not, it might be less confusing to tell Circe to deserialize null as an empty list. (I can help with that.)

Reply:
That's a fair point. I can't, off the top of my head, think of a use-case where the distinction is necessary. I welcome any help on the `Circe` front to update the deserialization to what you describe.

Reply:
Added a new decoder to handle cases where a field we expect to contain a list is either `null` or non-existent. Also converted all instances of `Seq` to `List`.